src/main/java/com/google/api/codegen/transformer/go/GoSurfaceNamer.java

```diff
@@ -0,0 +1,202 @@
+    if (s.length() > 0) {
+      s = Character.toLowerCase(s.charAt(0)) + s.substring(1);
+    }
+    return s;
+  }
+
+  @Override
+  public String getGrpcClientTypeName(Interface service) {
+    return converter.getTypeName(service).getNickname() + "Client";
+  }
+
+  @Override
+  public String getGrpcClientConstructorName(Interface service) {
+    return getGrpcClientTypeName(service).replace(".", ".New");
+  }
+
+  @Override
+  public String getCallSettingsTypeName(Interface service) {
+    return className(clientNamePrefix(service).join("call").join("options"));
+  }
+
+  @Override
+  public String getDefaultApiSettingsFunctionName(Interface service) {
+    return Name.from("default")
+        .join(clientNamePrefix(service))
+        .join("client")
+        .join("options")
+        .toLowerCamel();
+  }
+
+  @Override
+  public String getDefaultCallSettingsFunctionName(Interface service) {
+    return Name.from("default")
+        .join(clientNamePrefix(service))
+        .join("call")
+        .join("options")
+        .toLowerCamel();
+  }
+
+  @Override
+  public String getApiWrapperClassName(Interface service) {
+    return className(clientNamePrefix(service).join("client"));
+  }
+
+  @Override
+  public String getApiWrapperClassConstructorName(Interface service) {
+    return methodName(Name.from("new").join(clientNamePrefix(service)).join("client"));
+  }
+
+  @Override
+  public String getApiWrapperClassConstructorExampleName(Interface service) {
+    return methodName(
+        Name.from("example").join("new").join(clientNamePrefix(service)).join("client"));
+  }
+
+  @Override
+  public String getApiMethodExampleName(Interface service, Method method) {
+    return methodName(Name.from("example").join(clientNamePrefix(service)).join("client"))
+        + "_"
+        + getApiMethodName(method);
+  }
+
+  @Override
+  public String getLocalPackageName() {
+    // packagePath is in form "cloud.google.com/go/library/apiv1";
+    // we want "library".
+    String[] parts = packagePath.split("/");
+    return parts[parts.length - 2];
+  }
+
+  private Name clientNamePrefix(Interface service) {
+    String name = getReducedServiceName(service);
+    // If there's only one service, or the service name matches the package name, don't prefix with
+    // the service name.
+    if (model.getSymbolTable().getInterfaces().size() == 1 || name.equals(getLocalPackageName())) {
+      name = "";
+    }
+    return Name.upperCamel(name);
+  }
+
+  /**
+   * Returns the service name with common suffixes removed.
+   *
+   * For example:
+   *  "LoggingServiceV2" becomes "logging"
+   */
+  public static String getReducedServiceName(Interface service) {
+    String name = service.getSimpleName().replaceAll("V[0-9]+$", "");
+    name = name.replaceAll("Service$", "");
+    return Name.upperCamel(name).toLowerUnderscore();
+  }
+
+  public String getStatusCodeName(Status.Code code) {
+    return Name.upperUnderscore(code.toString()).toUpperCamel();
```

Comment:
What type of "thing" is a status code in Go - is it an enum or something like that?